src/targets/gpu/include/migraphx/gpu/compiler.hpp

```diff
@@ -45,18 +45,27 @@ struct compiler_replace
 {
     compiler_replace() = default;
 
-    compiler_replace(const operation& op) : code_object{op} {}
+    compiler_replace(const operation& op) : code_object{{op}} {}
 
     template <class F>
     compiler_replace(const operation& op, F f)
+        : code_object{{op}},
+          replace_fn([=](const compiler_replace& cr, module& m, instruction_ref ins) {
+              f(m, ins, cr.code_object.front());
+          })
+    {
+    }
+
+    template <class F>
+    compiler_replace(const std::vector<operation>& op, F f)
         : code_object{op},
           replace_fn([=](const compiler_replace& cr, module& m, instruction_ref ins) {
               f(m, ins, cr.code_object);
           })
     {
     }
 
-    operation code_object = {};
+    std::vector<operation> code_object = {};
```

Comment:
This should be renamed to `code_objects` as well to make it clear it can be multiple code objects.